File: helixtrajectoryj/src/main/java/org/team2363/helixtrajectory/SwerveModule.java
package org.team2363.helixtrajectory;


public final class SwerveModule {

    public final double x;
    public final double y;
    public final double wheelRadius;
    public final double wheelMaxAngularVelocity;
    public final double wheelMaxTorque;

    public SwerveModule(double x, double y, double wheelRadius, double wheelMaxAngularVelocity, double wheelMaxTorque) {
        this.x = x;
        this.y = y;
        this.wheelRadius = wheelRadius;
        this.wheelMaxAngularVelocity = wheelMaxAngularVelocity;
        this.wheelMaxTorque = wheelMaxTorque;
    }

    @Override
    public String toString() {
        return "{\"x\": " + x
            + ", \"y\": " + y
            + ", \"wheel_radius\": " + wheelRadius
            + ", \"wheel_max_angular_velocity\": " + wheelMaxAngularVelocity
            + ", \"wheel_max_torque\": " + wheelMaxTorque
            + "}";
    }
}

File: helixtrajectorycpp/src/cpp/drivetrain/SwerveModule.cpp
// Copyright (c) TrajoptLib contributors

#include "drivetrain/SwerveModule.h"

#include <iostream>

namespace helixtrajectory {

SwerveModule::SwerveModule(double x, double y, double wheelRadius,
                           double wheelMaxAngularVelocity,
                           double wheelMaxTorque)
    : x(x),
      y(y),
      wheelRadius(wheelRadius),
      wheelMaxAngularVelocity(wheelMaxAngularVelocity),
      wheelMaxTorque(wheelMaxTorque) {}

std::ostream& operator<<(std::ostream& stream, const SwerveModule& module) {
  return stream << "{\"x\": " << module.x << ", \"y\": " << module.y
                << ", \"wheel_radius\": " << module.wheelRadius
                << ", \"wheel_max_angular_velocity\": "
                << module.wheelMaxAngularVelocity
                << ", \"wheel_max_torque\": " << module.wheelMaxTorque << "}";
}
}  // namespace helixtrajectory
